ADAL/src/ADAuthenticationResult+Internal.m

```diff
@@ -155,20 +158,48 @@ + (ADAuthenticationResult*)resultForBrokerErrorResponse:(NSDictionary*)response
     
     // Otherwise parse out the error condition
     ADAuthenticationError* error = nil;
+    NSMutableDictionary* userInfo = nil;
     
     NSString* errorDetails = [response valueForKey:OAUTH2_ERROR_DESCRIPTION];
     if (!errorDetails)
     {
         errorDetails = @"Broker did not provide any details";
     }
+
+    if (response[BROKER_APP_VERSION])
+    {
+        [userInfo setValue:response[BROKER_APP_VERSION] forKey:BROKER_APP_VERSION];
+    }
         
     NSString* strErrorCode = [response valueForKey:@"error_code"];
     NSInteger errorCode = AD_ERROR_TOKENBROKER_UNKNOWN;
     if (strErrorCode && ![strErrorCode isEqualToString:@"0"])
     {
         errorCode = [strErrorCode integerValue];
     }
+
     
+
+    if (errorCode == AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED)
+    {
+        // For protection_policy_required error, add extra info for the app in the userInfo dictionary of the error
+        userInfo = [[NSMutableDictionary alloc] initWithCapacity:3];
+        if (response[@"suberror"])
+        {
+            [userInfo setValue:response[@"suberror"] forKey:@"suberror"];
+        }
+
+        if (response[@"userId"])
```

Comment:
@jasoncoolmax will this ever be filled if broker call fails?

Reply:
Mike is making changes about the user id part. 
Yes, broker should return user id together with `AD_ERROR_SERVER_PROTECTION_POLICY_REQUIRED`, even broker fails to get the mam tokens.

BTW, here the key should be `user_id` but not `userId`, according to the PR in broker repo :)